wdl-cli/src/inputs.rs

```diff
@@ -0,0 +1,507 @@
+    }
+
+    #[test]
+    fn coalesce() {
+        // Helper functions.
+        fn check_string_value(inputs: &Inputs, key: &str, value: &str) {
+            let (_, input) = inputs.get(key).unwrap();
+            assert_eq!(input.as_string().unwrap().as_str(), value);
+        }
+
+        fn check_float_value(inputs: &Inputs, key: &str, value: f64) {
+            let (_, input) = inputs.get(key).unwrap();
+            assert_eq!(input.as_float().unwrap(), value);
+        }
+
+        fn check_boolean_value(inputs: &Inputs, key: &str, value: bool) {
+            let (_, input) = inputs.get(key).unwrap();
+            assert_eq!(input.as_boolean().unwrap(), value);
+        }
+
+        fn check_integer_value(inputs: &Inputs, key: &str, value: i64) {
+            let (_, input) = inputs.get(key).unwrap();
+            assert_eq!(input.as_integer().unwrap(), value);
+        }
+
+        // The standard coalescing order.
+        let inputs = Inputs::coalesce([
+            "./tests/fixtures/inputs_one.json",
+            "./tests/fixtures/inputs_two.json",
+            "./tests/fixtures/inputs_three.yml",
+        ])
+        .unwrap();
+
+        assert_eq!(inputs.len(), 5);
+        check_string_value(&inputs, "foo", "bar");
+        check_float_value(&inputs, "baz", 128.0);
+        check_string_value(&inputs, "quux", "qil");
+        check_string_value(&inputs, "new", "foobarbaz");
+        check_string_value(&inputs, "new_two", "bazbarfoo");
+
+        // The opposite coalescing order.
+        let inputs = Inputs::coalesce([
+            "./tests/fixtures/inputs_three.yml",
+            "./tests/fixtures/inputs_two.json",
+            "./tests/fixtures/inputs_one.json",
+        ])
+        .unwrap();
+
+        assert_eq!(inputs.len(), 5);
+        check_string_value(&inputs, "foo", "bar");
+        check_float_value(&inputs, "baz", 42.0);
+        check_string_value(&inputs, "quux", "qil");
+        check_string_value(&inputs, "new", "foobarbaz");
+        check_string_value(&inputs, "new_two", "bazbarfoo");
+
+        // An example with some random key-value pairs thrown in.
+        let inputs = Inputs::coalesce([
+            r#"sandwich=-100"#,
+            "./tests/fixtures/inputs_one.json",
+            "./tests/fixtures/inputs_two.json",
+            r#"quux="jacks""#,
+            "./tests/fixtures/inputs_three.yml",
+            r#"baz=false"#,
+        ])
+        .unwrap();
+
+        assert_eq!(inputs.len(), 6);
+        check_string_value(&inputs, "foo", "bar");
+        check_boolean_value(&inputs, "baz", false);
+        check_string_value(&inputs, "quux", "jacks");
+        check_string_value(&inputs, "new", "foobarbaz");
+        check_string_value(&inputs, "new_two", "bazbarfoo");
+        check_integer_value(&inputs, "sandwich", -100);
+
+        // An invalid key-value pair.
+        let error =
+            Inputs::coalesce(["./tests/fixtures/inputs_one.json", "foo=baz#bar"]).unwrap_err();
```

Comment:
would this work `baz#bar` were wrapped in quotes?

Reply:
Yes, it would.